### PR TITLE
feat(tracking): first-class status lanes + library export (#165)

### LINKED ISSUE #165: [Pillar] Core tracking experience for read/currently-reading/want-to-read
## Objective
Make bibliomnomnom the easiest place to maintain three accurate lists:
- read
- currently-reading
- want-to-read

## Why this matters
If this is not frictionless, everything else (voice notes, profile, recommendations) becomes low-trust or low-data.

## Success criteria
- Status updates are instant and reliable from library + detail views.
- Users can quickly find and manage each list.
- Import/export and discovery reduce manual entry burden.

## In-scope child issues
- [ ] #169 Tracking UX: first-class read/currently-reading/want-to-read lanes
- [ ] #36 External book search (Google Books API)
- [ ] #34 Export library to JSON/CSV/Markdown
- [ ] #32 Queue sorting: date, alphabetical, shuffle
- [ ] #60 CMD+K command palette for library search
- [ ] #56 Improve form validation and error messages
- [ ] #54 Consolidate book filtering to single pass

## Explicitly out of scope for this pillar
- Monetization experiments
- Social launch tasks
- Design-system polish not tied to list-management speed

### PR DESCRIPTION
**Pull request by phrazzld**
## Summary

Closes #165

Delivers two foundational features from the Core Tracking pillar — the product promise that read/currently-reading/want-to-read are easy to maintain:

1. **First-class tracking lanes** (#169): The library now surfaces three explicit tabs instead of a generic "Library/Queue" split. Counts per lane are always visible. Every card in a lane shows a one-click action to advance the book to its natural next state (e.g., "Start reading" on Want to Read, "Mark as read" on Currently Reading).

2. **Library export** (#34): Users can export their full library as JSON, CSV, or Markdown. Addresses lock-in anxiety and lays groundwork for GDPR/CCPA compliance.

Together these make the three core lists the dominant UI surface, and give users data portability from day one.

## Changes

**Tracking lanes** (#169):
- `components/book/BookGrid.tsx` — Replaced `"library" | "to-read"` filter with `"currently-reading" | "read" | "want-to-read" | "favorites"`. Default tab is Currently Reading. Per-status counts always visible in tab labels.
- `components/book/BookTile.tsx` — Added inline status-action button per card. Status-aware: "Start reading" (want-to-read → currently-reading), "Mark as read" (currently-reading → read). Click is non-navigating (stopPropagation). Fires `books.updateStatus` mutation directly with optimistic update.
- `components/book/__tests__/BookTile.test.tsx` — 109 lines of behavior tests covering status action rendering, click-to-advance, navigation isolation, and no-action for "read" books.

**Export** (#34):
- `convex/books.ts` — Added `exportAllData` query returning `{ version, exportedAt, books, notes }` for the authenticated user.
- `lib/export.ts` — Pure format converters: `toBooksJSON`, `toBooksCSV` (Goodreads-compatible headers), `toBooksMarkdown` (one section per status, Obsidian-friendly).
- `lib/__tests__/export.test.ts` — 142 lines covering all three formats, special characters, empty libraries, and field escaping.
- `components/library/ExportButton.tsx` — Dropdown trigger in library header. Calls `exportAllData`, converts client-side, triggers browser download. No server round-trip for file generation.

## Acceptance Criteria

From #169:
- [x] Library has clear sections / tabs for Read, Currently Reading, Want to Read
- [x] Moving a book between states takes one interaction from list cards
- [x] Counts are always visible for each state
- [x] Filtering/sorting preserves state context

From #34:
- [x] Export button accessible from library
- [x] Format selection: JSON, CSV, Markdown
- [x] JSON includes full data with metadata
- [x] CSV is Goodreads-compatible
- [x] Markdown is human-readable / Obsidian-compatible

## Manual QA

**Tracking lanes:**
1. `bun run dev` → sign in → navigate to `/library`
2. Verify four tabs: Currently Reading / Read / Want to Read / Favorites — each with count badges
3. Add or find a book in "Want to Read" → confirm "Start reading" button appears on the tile
4. Click "Start reading" → book should disappear from Want to Read tab and appear under Currently Reading (real-time, no reload)
5. On a Currently Reading book, click "Mark as read" → moves to Read tab
6. Books already in Read state should show no action button
7. Confirm tile click still navigates to detail page (action button doesn't trigger navigation)

**Export:**
1. In `/library`, click the Export dropdown (top-right toolbar)
2. Select "JSON" → `library-export.json` downloads; verify `version`, `exportedAt`, `books[]`, `notes[]` fields
3. Select "CSV" → `library-export.csv` downloads; open in spreadsheet, verify Goodreads-compatible column headers
4. Select "Markdown" → `library-export.md` downloads; verify books grouped by status, readable prose
5. Test with empty library → each format should produce valid empty output (no crash)

## Before / After

**Before:**
- Library showed a combined "Library" tab (all read + currently-reading books together) and a separate "Queue" tab for want-to-read
- Changing a book's status required opening the detail view or using a dropdown — multiple clicks
- No way to export data; users were locked in

**After:**
- Three explicit status lanes as primary navigation: Currently Reading, Read, Want to Read (plus Favorites)
- Per-lane counts always visible in tab labels
- One-click status advancement directly from library cards (no navigation required)
- Export dropdown in library header: JSON / CSV / Markdown download with a single click

## Test Coverage

| File | Tests | What's covered |
|------|-------|---------------|
| `components/book/__tests__/BookTile.test.tsx` | ~12 | Status action rendering per status, click advances status, stopPropagation, no action on "read" |
| `lib/__tests__/export.test.ts` | ~18 | All three formats, field escaping, empty library, special characters in titles/authors |

**Gaps:**
- `BookGrid.tsx` tab-switching behavior — not unit tested (integration/E2E candidate)
- `ExportButton.tsx` — not tested; it's a thin wrapper over `export.ts` which is fully tested
- No E2E test for the full status-transition flow; manual QA covers this for now

## Summary by CodeRabbit

* **New Features**
  * Added export functionality supporting JSON, CSV, and Markdown formats for your library
  * Added quick-action buttons on book tiles to transition between reading statuses (Start Reading, Mark as Read)

* **Improvements**
  * Reorganized book filters: Currently Reading, Read, Want to Read, and Favorites
  * Updated empty states and messaging throughout the app
  * Streamlined interface by removing onboarding flow

* **Tests**
  * Added test coverage for status action functionality and export formats

---

## Polish Pass

**Before:** Working implementation with duplicated test factories, DOM-dependent side-effect in a pure formatter module, and missing edge-case test coverage for CSV escaping and empty-library edge cases.

**After:**

**Refactoring:**
- Extracted shared test fixtures (`makeBook`, `makeNote`, `fakeBookId`, etc.) into `lib/__tests__/fixtures.ts` — eliminates duplication between BookTile and export tests, single point of update when schema changes
- Moved `downloadFile` from `lib/export.ts` to `lib/download.ts` — keeps the three format converters (`toJSON`, `toCSV`, `toMarkdown`) pure (no DOM dependency), making them SSR-safe and trivially testable

**Test Coverage (+7 tests):**
- Empty library: all three formats produce valid output (no crashes on `[]`)
- CSV escaping: double quotes in title/author correctly doubled per RFC 4180
- CSV quoting: commas in values preserved inside double-quoted fields
- Markdown: books without notes omit the Notes section cleanly
- Markdown: `reflection` note type renders as plain text (documents current behavior)

**Issues Created:**
- #185: Remove unused `status`/`favoritesOnly` args from `books.list` query (dead code after client-side filtering move)
- #186: Include listening sessions in library export for complete data portability

**Quality Gates:** typecheck, lint, 463/463 tests — all green.

### CODE REVIEW

File: lib/export.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  "ISBN",
+  "My Rating",
+  "Date Read",
+  "Date Added",
+  "Bookshelves",
+  "My Review",
+];
+
+function formatDateYmd(timestamp: number): string {
+  const [formatted] = new Date(timestamp).toISOString().split("T");
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply lib/export.ts]: Fixed in 14814ed. Replaced `toISOString().split('T')` UTC truncation with explicit local-time components (`getFullYear`/`getMonth`/`getDate`), so exported dates match the user's calendar regardless of timezone.

File: convex/books.ts [hunk 2]
```diff
@@ -65,6 +65,22 @@ export const list = query({
...
+    const books = await ctx.db
+      .query("books")
+      .withIndex("by_user", (q) => q.eq("userId", userId))
+      .collect();
+    const notes = await ctx.db
+      .query("notes")
+      .withIndex("by_user", (q) => q.eq("userId", userId))
+      .collect();
+
+    return { version: "1.0", exportedAt: Date.now(), books, notes };
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply convex/books.ts]: Fixed in 14814ed. Removed `exportedAt` from the Convex query evaluation path — it's now stamped in `ExportButton.handleExport` at click time via `{ ...exportData, exportedAt: Date.now() }`. Metadata always reflects the actual download moment.

File: components/book/BookTile.tsx [hunk 3]
```diff
@@ -45,6 +53,44 @@ export function BookTile({ book }: BookTileProps) {
...
   };
 
+  const handleStatusActionClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.stopPropagation();
+    if (!statusAction) {
+      return;
+    }
+
+    void updateStatus({
+      id: book._id,
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply components/book/BookTile.tsx]: Fixed in 14814ed — replaced `.catch(() => undefined)` with a destructive toast. Mutation error path now covered in the test suite (8f3ca3d).

**Comment by phrazzld** [comment]
## PR Unblocked

**Conflicts**: none

**CI**: green (latest run in progress on new commits; pre-push hooks passed locally)

**Reviews**: 7 threads — 6 fixed, 1 deferred

| Thread | File | Fix |
|--------|------|-----|
| Silent error swallowing (CodeRabbit) | `BookTile.tsx:66` | Added destructive toast on mutation failure (14814ed) |
| UTC date truncation (Codex) | `lib/export.ts:22` | Local-time formatting via `getFullYear/Month/Date` (14814ed) |
| Stale exportedAt timestamp (Codex) | `convex/books.ts:80` | Stamp `Date.now()` at click time in ExportButton (14814ed) |
| Silent error (Cerberus duplicate) | `BookTile.tsx:65` | Same fix as above (14814ed) |
| Missing test coverage (Cerberus) | `BookTile.tsx:94` | Added toast-error and keyboard stopPropagation tests (8f3ca3d) |
| Eager export fetch (Cerberus) | `ExportButton.tsx:22` | `useQuery(..., "skip")` until menu opens (8f3ca3d) |
| Unbounded `.collect()` (Cerberus) | `convex/books.ts:72` | **Deferred** → #184 (pagination required; arbitrary truncation would silently lose data) |

Ready for re-review.
